test/hotspot/jtreg/compiler/lib/template_framework/NameSet.java

```diff
@@ -84,6 +84,9 @@ public List<Name> toList(Predicate predicate) {
     public Name sample(Predicate predicate) {
         long w = weight(predicate);
         if (w <= 0) {
+            if (w < 0) {
+                throw new RuntimeException("Negative weight not allowed: " + w);
+            }
```

Comment:
I thought zero is also not allowed?

Reply:
As asked for offline: added some more comments here.